### ISSUE TITLE
off-by-one span in doc_markdown warning on single tick

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#2481]
clap-rs has code like this:
````rust

/// `Arg::required(true)` such as `--config <FILE>` = `Arg::value_name("FILE")` and
/// `Arg::required(true)

````
and clippy is confused by the missing(?) trailing "`":

````
warning: you should put `Arg::required(true` between ticks in the documentation
 --> src/main.rs:2:6
  |
2 | /// `Arg::required(true)
  |      ^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(doc_markdown)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#doc_markdown

````
```` `Arg::required(true` ```` is not what we want I think... :)

source code: https://github.com/kbknapp/clap-rs/blob/03e413d7175d35827cd7d8908d47dbae15a849a3/src/macros.rs#L610

**Comment by mockersf** [comment]
related to #2210

**Comment by phansch** [comment]
Triage: This is still happening: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=1c3ef693dbb64babf73fcf948f0d9f0d

**Comment by giraffate** [comment]
This was fixed by https://github.com/rust-lang/rust-clippy/pull/7357, so I'm closing this.
